framework/src/main/java/org/checkerframework/framework/type/AsSuperVisitor.java

```diff
@@ -803,9 +798,16 @@ public AnnotatedTypeMirror visitWildcard_Wildcard(
             isUninferredTypeAgrument = true;
             superType.setUninferredTypeArgument();
         }
-        AnnotatedTypeMirror upperBound =
-                visit(type.getExtendsBound(), superType.getExtendsBound(), p);
-        superType.setExtendsBound(upperBound);
+        if (types.isSubtype(
+                type.getExtendsBound().getUnderlyingType(),
+                superType.getExtendsBound().getUnderlyingType())) {
+            AnnotatedTypeMirror upperBound =
+                    visit(type.getExtendsBound(), superType.getExtendsBound(), p);
+            superType.setExtendsBound(upperBound);
+        } else {
+            copyPrimaryAnnos(type.getExtendsBound(), superType.getExtendsBound());
```

Comment:
Could you also add a comment here why this is the correct thing to do? Why copy anything if the underlying types are not subtypes?